sky/backends/backend_utils.py

```diff
@@ -816,6 +659,36 @@ def _remove_multinode_config(
             if 'Host' in config_str:
                 break
 
+    @classmethod
+    @timeline.FileLockEvent(ssh_conf_lock_path)
```

Comment:
nit: the lock here might be too coarse grained, when we remove the backward compatibility, as we may only need a lock per cluster then. We can probably add a TODO.

Reply:
Yeah! I was thinking about asking that. Will do!